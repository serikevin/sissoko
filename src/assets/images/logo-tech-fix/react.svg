<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 60.9 54.2"><defs><style>.cls-1{fill:#fff;}</style></defs><path class="cls-1" d="M130.7,26.9c0-4-5.1-7.9-12.8-10.2,1.7-7.9,1-14.2-2.5-16.2a5.3,5.3,0,0,0-2.8-.7V2.6a3.2,3.2,0,0,1,1.4.3c1.7,1,2.4,4.6,1.8,9.4a22.5,22.5,0,0,1-.6,3.6,59.2,59.2,0,0,0-7.9-1.3,54.4,54.4,0,0,0-5.1-6.2c4-3.8,7.8-5.8,10.4-5.8V-.2c-3.4,0-7.9,2.4-12.4,6.6-4.5-4.1-9-6.6-12.4-6.6V2.6c2.6,0,6.4,2.1,10.4,5.8a67.6,67.6,0,0,0-5.1,6.2,59.2,59.2,0,0,0-7.9,1.3,22.5,22.5,0,0,1-.6-3.6c-.6-4.7.1-8.4,1.8-9.4a2.9,2.9,0,0,1,1.4-.3V-.1A5.3,5.3,0,0,0,85,.6c-3.5,2-4.3,8.2-2.5,16.1-7.7,2.4-12.7,6.2-12.7,10.2s5,7.9,12.8,10.2c-1.8,7.9-1,14.2,2.5,16.2a5.1,5.1,0,0,0,2.8.7c3.4,0,7.9-2.4,12.4-6.7,4.5,4.2,9,6.6,12.4,6.6a5.5,5.5,0,0,0,2.8-.7c3.5-2,4.2-8.2,2.4-16.1,7.7-2.4,12.8-6.2,12.8-10.2Zm-16.2-8.3a47.5,47.5,0,0,1-1.7,4.9c-.5-1-1-2-1.6-3l-1.8-2.9,5.1,1Zm-5.7,13.2a38.8,38.8,0,0,1-3,4.8l-5.6.2-5.6-.2-3-4.7c-.9-1.7-1.8-3.3-2.5-5s1.6-3.3,2.5-4.9,2-3.3,3-4.8l5.6-.2,5.6.2q1.5,2.3,3,4.8c.9,1.6,1.8,3.2,2.6,4.9l-2.6,4.9Zm4-1.6a32.8,32.8,0,0,1,1.7,5l-5.1,1,1.8-3c.6-1,1.1-2,1.6-3ZM100.2,43.5c-1.1-1.2-2.3-2.6-3.4-4h6.9C102.5,40.9,101.4,42.3,100.2,43.5ZM91,36.2l-5.1-1c.5-1.6,1.1-3.3,1.7-4.9l1.6,3Zm9.2-25.9c1.1,1.2,2.3,2.6,3.4,4H96.8A44.7,44.7,0,0,1,100.2,10.3ZM91,17.6l-1.8,3-1.6,3c-.7-1.7-1.2-3.4-1.7-5l5.1-1ZM79.8,33.2c-4.4-1.9-7.2-4.4-7.2-6.3s2.8-4.4,7.2-6.3l3.4-1.2A70.1,70.1,0,0,0,86,26.9a52.6,52.6,0,0,0-2.7,7.5,28.6,28.6,0,0,1-3.5-1.2Zm6.7,17.7c-1.7-1-2.5-4.7-1.9-9.4.1-1.2.4-2.4.6-3.6a62,62,0,0,0,7.9,1.3,54.9,54.9,0,0,0,5.2,6.2c-4.1,3.8-7.9,5.8-10.4,5.8a2.6,2.6,0,0,1-1.4-.3Zm29.4-9.5c.6,4.8-.2,8.5-1.8,9.5a3.3,3.3,0,0,1-1.5.3c-2.5,0-6.3-2.1-10.4-5.8a45.6,45.6,0,0,0,5.1-6.2,58.8,58.8,0,0,0,7.9-1.4C115.5,39.1,115.7,40.3,115.9,41.4Zm4.8-8.2-3.5,1.2a53.6,53.6,0,0,0-2.8-7.5,53.6,53.6,0,0,0,2.8-7.5,35.9,35.9,0,0,1,3.5,1.2c4.4,1.9,7.2,4.4,7.2,6.3s-2.9,4.4-7.2,6.3Z" transform="translate(-69.8 0.2)"/><circle class="cls-1" cx="30.4" cy="27.1" r="5.7"/></svg>