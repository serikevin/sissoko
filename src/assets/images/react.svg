<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 60.9 54.2"><defs><style>.cls-1{fill:#00d8ff;}</style></defs><path class="cls-1" d="M60.9,27.1c0-4-5.1-7.9-12.8-10.2C49.8,9,49.1,2.7,45.6.7A5.5,5.5,0,0,0,42.8,0V2.8a3.2,3.2,0,0,1,1.4.3c1.7,1,2.4,4.6,1.8,9.4a22.5,22.5,0,0,1-.6,3.6,68.5,68.5,0,0,0-7.9-1.3,43.1,43.1,0,0,0-5.1-6.2c4-3.8,7.8-5.8,10.4-5.8V0c-3.4,0-7.9,2.4-12.4,6.6C25.9,2.5,21.4,0,18,0V2.8c2.6,0,6.4,2.1,10.4,5.8a76.4,76.4,0,0,0-5.1,6.2,68.5,68.5,0,0,0-7.9,1.3,22.5,22.5,0,0,1-.6-3.6c-.6-4.7.1-8.4,1.8-9.4A3.2,3.2,0,0,1,18,2.8V.1a5.5,5.5,0,0,0-2.8.7c-3.5,2-4.3,8.2-2.5,16.1C5,19.3,0,23.1,0,27.1S5,35,12.8,37.3c-1.8,7.9-1,14.2,2.5,16.2a5.3,5.3,0,0,0,2.8.7c3.4,0,7.9-2.4,12.4-6.7,4.5,4.2,9,6.6,12.4,6.6a5.5,5.5,0,0,0,2.8-.7c3.5-2,4.2-8.2,2.4-16.1,7.7-2.4,12.8-6.2,12.8-10.2ZM44.7,18.8A40.5,40.5,0,0,1,43,23.7c-.5-1-1-2-1.6-3l-1.8-2.9ZM39,32a38.8,38.8,0,0,1-3,4.8l-5.6.2-5.6-.2-3-4.7c-.9-1.7-1.8-3.3-2.5-5s1.6-3.3,2.5-4.9,2-3.3,3-4.8l5.6-.2,5.6.2q1.5,2.3,3,4.8c.9,1.6,1.8,3.2,2.6,4.9Zm4-1.6a29.5,29.5,0,0,1,1.7,5l-5.1,1,1.8-3C42,32.4,42.5,31.4,43,30.4ZM30.4,43.7c-1.1-1.2-2.3-2.6-3.4-4h6.9C32.7,41.1,31.6,42.5,30.4,43.7Zm-9.2-7.3-5.1-1c.5-1.6,1.1-3.3,1.7-4.9l1.6,3Zm9.2-25.9c1.1,1.2,2.3,2.6,3.4,4H27C28.1,13.1,29.2,11.8,30.4,10.5Zm-9.2,7.3-1.8,3-1.6,3c-.7-1.7-1.2-3.4-1.7-5ZM10,33.4C5.6,31.5,2.8,29,2.8,27.1s2.8-4.4,7.2-6.3l3.4-1.2a60.9,60.9,0,0,0,2.8,7.5,47.8,47.8,0,0,0-2.7,7.5A20.6,20.6,0,0,1,10,33.4Zm6.7,17.7c-1.7-1-2.5-4.7-1.9-9.4.1-1.2.4-2.4.6-3.6a68.5,68.5,0,0,0,7.9,1.3,51.7,51.7,0,0,0,5.2,6.2c-4.1,3.8-7.9,5.8-10.4,5.8A3.2,3.2,0,0,1,16.7,51.1Zm29.4-9.5c.6,4.8-.2,8.5-1.8,9.5a4,4,0,0,1-1.5.3c-2.5,0-6.3-2.1-10.4-5.8a43.1,43.1,0,0,0,5.1-6.2A68.6,68.6,0,0,0,45.4,38C45.7,39.3,45.9,40.5,46.1,41.6Zm4.8-8.2-3.5,1.2a48.1,48.1,0,0,0-2.8-7.5,48.1,48.1,0,0,0,2.8-7.5,20.6,20.6,0,0,1,3.5,1.2c4.4,1.9,7.2,4.4,7.2,6.3S55.2,31.5,50.9,33.4Z" transform="translate(0 0)"/><circle class="cls-1" cx="30.4" cy="27.1" r="5.7"/></svg>