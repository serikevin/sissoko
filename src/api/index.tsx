import axios from 'axios';
import { SERVER_BASE_URL } from 'constant/url';
import * as Sentry from "@sentry/react";

const client = axios.create({
    baseURL: SERVER_BASE_URL,
    timeout: 1000,
    headers: {
        'Content-Type': 'application/json',
    },
});

export const getAbout = async () => {
    try {
        const response = await client.get("/about");
        return response.data
    } catch (error) {
        Sentry.captureException(new Error("500 - Internal Server Error - Cant get about data"));
    }
};

export const getPortfolioList = async (page : number = 1) => {
    try {
        const response = await client.get("/portofolio", {
            params: {
                page: page,
                limit: 9
            }
        });
        return response.data
    } catch (error) {
        Sentry.captureException(new Error("500 - Internal Server Error - Cant get about data"));
    }
};

export const getPortfolioDetail = async (id : number) => {
    try {
        const response = await client.get("/portofolio/" + id);
        return response.data
    } catch (error) {
        Sentry.captureException(new Error("500 - Internal Server Error - Cant get about data"));
    }
};

export const getCertificateList = async (page : number = 1) => {
    try {
        const response = await client.get("/certificate", {
            params: {
                page: page,
                limit: 9
            }
        });
        return response.data
    } catch (error) {
        Sentry.captureException(new Error("500 - Internal Server Error - Cant get about data"));
    }
};