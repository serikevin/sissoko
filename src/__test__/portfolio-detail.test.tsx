import { render } from '@testing-library/react';
import { getPortfolioDetail } from 'api';
import Detail from 'pages/Detail';

test('Fetch portfolio detail with correct id', async () => {
	const { data } = await getPortfolioDetail(1);
	expect(data.id).toBe(1);
});

test('Fetch portfolio detail with wrong id', async () => {
	// eslint-disable-next-line jest/no-conditional-expect
	return getPortfolioDetail(-1).catch(e => expect(e).toMatch('error'));
});

test('Render check after async', async () => {
	try {
		await getPortfolioDetail(1);
		const { container } = render(<Detail />);
		expect(container).toMatchSnapshot()
	} catch (e) {
		// eslint-disable-next-line jest/no-conditional-expect
		expect(e).toMatch('error');
	}
});