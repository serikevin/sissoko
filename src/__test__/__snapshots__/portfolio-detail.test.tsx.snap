// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render check after async 1`] = `
<div>
  <nav
    class="navbar nav-down"
    data-animation="shrink"
    data-fullwidth="true"
    data-menu-style="transparent"
  >
    <div
      class="container bg-dark"
    >
      <div
        class="navbar-header bg-dark"
      >
        <div
          class="container"
        >
          <button
            aria-controls="navbar"
            aria-expanded="false"
            class="navbar-toggle collapsed"
            data-target="#navbar"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="sr-only"
            >
              Toggle navigation
            </span>
            <span
              class="icon-bar top-bar"
            />
            <span
              class="icon-bar middle-bar"
            />
            <span
              class="icon-bar bottom-bar"
            />
          </button>
          <a
            class="navbar-brand to-top"
            href="/#"
          >
            <img
              alt="#"
              class="logo-nav logo-size-navbar"
              src="app-logo.svg"
            />
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse "
      >
        <div
          class="container"
        >
          <ul
            class="nav navbar-nav menu-right"
          >
            <li>
              <a
                href="curriculum-vitae.pdf"
                rel="noreferrer"
                target="_blank"
              >
                Resume
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/yovie-alfa-guistuta-b02496146"
                rel="noreferrer"
                target="_blank"
              >
                Linkedin
              </a>
            </li>
            <li>
              <a
                href="https://www.instagram.com/yoviealfa"
                rel="noreferrer"
                target="_blank"
              >
                Instagram
              </a>
            </li>
            <li>
              <a
                href="https://github.com/yoviealfaguistuta"
                rel="noreferrer"
                target="_blank"
              >
                Github
              </a>
            </li>
            <li
              class="btn-main-header"
            >
              <a
                class="nopadding"
                href="mailto:yoviealfaguistuta@gmail.com"
              >
                Contact Me
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
  <section
    class="bg-dark pb50 text-left"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <svg
          aria-labelledby="kvugdh-aria"
          role="img"
          viewBox="0 0 700 300"
        >
          <title
            id="kvugdh-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#kvugdh-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#kvugdh-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="kvugdh-diff"
            >
              <rect
                height="32"
                rx="14"
                ry="14"
                width="72"
                x="376"
                y="253"
              />
              <rect
                height="416"
                rx="16"
                ry="16"
                width="274"
                x="55"
                y="42"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="101"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="121"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="81"
              />
              <rect
                height="29"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="41"
              />
              <rect
                height="29"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="140"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="180"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="100"
                x="376"
                y="200"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="220"
              />
            </clippath>
            <lineargradient
              id="kvugdh-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#111111"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#000000"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#111111"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
    </div>
  </section>
  <footer
    class="site-footer bg-dark"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12 col-md-6 footer-space-about"
        >
          <h6>
            About
          </h6>
          Loading ...
        </div>
        <div
          class="col-lg-3 col-sm-3 col-md-3"
        >
          <h6>
            Sosial Media
          </h6>
          <ul
            class="footer-links "
          >
            <li>
              <a
                href="https://www.linkedin.com/in/yovie-alfa-guistuta-b02496146"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Linkedin
              </a>
            </li>
            <li>
              <a
                href="https://www.instagram.com/yoviealfa"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Instagram
              </a>
            </li>
            <li>
              <a
                href="https://github.com/yoviealfaguistuta"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Github
              </a>
            </li>
          </ul>
        </div>
        <div
          class="col-3 col-sm-3 col-md-3"
        >
          <h6>
            Contact
          </h6>
          <ul
            class="footer-links"
          >
            <li>
              <a
                href="https://t.me/yoviealfaguistuta"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Telegram
              </a>
            </li>
            <li>
              <a
                href="mailto:yoviealfaguistuta@gmail.com"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Email
              </a>
            </li>
            <li>
              <a
                href="https://www.google.com/maps/place/Jl.+Pulau+Sanama,+Way+Halim+Permai,+Way+Halim,+Kota+Bandar+Lampung,+Lampung+35132/@-5.3854168,105.2722535,15.79z/data=!4m5!3m4!1s0x2e40db2242631409:0x44c47ee3b4619d4e!8m2!3d-5.3851969!4d105.2741399"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="icon-footer"
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
                Address
              </a>
            </li>
          </ul>
        </div>
      </div>
      <hr
        class="small"
      />
    </div>
    <div
      class="container footer-container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="costum-center"
          >
            <div
              class="image-tech-footer"
            >
              <img
                alt="React JS"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/2300px-React-icon.svg.png"
                width="25"
              />
              <img
                alt="Go Languange"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/0/05/Go_Logo_Blue.svg/2560px-Go_Logo_Blue.svg.png"
                width="35"
              />
            </div>
            <div
              class="footer-tech text-desc"
            >
              This site using 
              <span
                class="text-highlight"
              >
                 React JS 
              </span>
              and 
              <span
                class="text-highlight"
              >
                Go Language
              </span>
            </div>
            <p
              class="res-foot text-desc"
            >
              Feel free to get the source code from my github 
              <a
                class="footer-link-github"
                href="https://github.com/yoviealfaguistuta/react-portofolio-frontend"
                rel="noreferrer"
                target="_blank"
              >
                Client Side
              </a>
               - 
              <a
                class="footer-link-github"
                href="https://github.com/yoviealfaguistuta/go-portofolio-api"
                rel="noreferrer"
                target="_blank"
              >
                Server Side
              </a>
            </p>
          </div>
        </div>
        <div
          class="col-md-12 col-sm-6 col-12 mb-0 pb-0 centered-footer-hosted border-top"
        >
          <p
            class="copyright-text text-desc"
          >
            Powered and Hosted by 
            <a
              href="https://www.digitalocean.com/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="Digital Ocean"
                class="digitalocean"
                src="https://static.cdnlogo.com/logos/d/54/digitalocean.svg"
              />
               
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
